File: Data Structures & Algorithms/task-scheduling/submission-1.py
class Solution:
    def leastInterval(self, tasks: List[str], n: int) -> int:
        m = len(tasks)
        count = Counter(tasks)
        maxHeap = [-cnt for cnt in count.values()]
        heapq.heapify(maxHeap)
        time = 0
        q = deque()
        while(maxHeap or q):
            if not maxHeap:
                time+=1
            if maxHeap:
                curr = heapq.heappop(maxHeap)
                time +=1
                if curr+1<0:
                    q.append([curr+1, time+n])
            while q and q[0][1]<=time:
                top = q.pop()
                maxHeap.append(top[0])
            heapq.heapify(maxHeap)
        return time

File: Data Structures & Algorithms/count-number-of-islands/submission-1.java
class Solution {
    public int numIslands(char[][] grid) {
        int count = 0;
        int n = grid.length;
        int m = grid[0].length;
        for (int i=0; i<n; i++){
            for (int j=0; j<m; j++){
                if(grid[i][j]=='1'){
                    fill(n,m,i,j,grid);
                    count+=1;
                }
            }
        }
        return count;


    }
    public void fill(int n, int m, int r, int c, char[][] grid){
        if (r>=0 && r<n && c>=0 && c<m && grid[r][c]=='1'){
            grid[r][c] = '0';
        }
        else{
            return;
        }
        fill(n,m,r+1,c,grid);
        fill(n,m,r-1,c,grid);
        fill(n,m,r,c-1,grid);
        fill(n,m,r,c+1,grid);
        return;
    }
}
